docs/serving/using-auto-tls.md

```diff
@@ -0,0 +1,150 @@
+---
+title: "Using Auto-TLS in Knative"
+weight: 15
+type: "docs"
+---
+
+Within Knative, we provide a feature of automatically provisioning and 
+configuring TLS certificates to terminate the external TLS connection. This doc
+provides a guide about how to enable and configure Auto TLS feature in Knative.
+
+## Prerequesties
+
+<!-- TODO(zhiminx) add the link about installing Istio with SDS enabled after PR https://github.com/knative/docs/pull/1272 is checked in-->
+1. Follow the [instructions]() to install Istio with SDS enabled.
+2. Follow the [instructions](../installing-cert-manager.md) to install Cert-Manager.
+
+## Configure Cert-Manager
+
+> Note that currently the Auto-TLS feature only supports DNS challenge.
+
+### Set up DNS challenge Provider
+
+Follow the [instructions](https://docs.cert-manager.io/en/latest/tasks/acme/configuring-dns01/index.html#supported-dns01-providers) to set up DNS challenge issuer according to the 
+DNS provider of your custom domain.
+
+#### Google Cloud DNS Provider Instruction
+
+Specifically for Google Cloud DNS, below are steps to set up the DNS challenge 
+issuer.
+
+##### Creating a Cloud DNS service account
+
+To add the TXT record, configure Knative with a service account that can be used
+by cert-manager to create and update the DNS record.
+
+To begin, create a new service account with the project role `dns.admin`:
+
+```shell
+# Set this to your GCP project ID
+export PROJECT_ID=<your-project-id>
```

Comment:
What do I need to do to associate my Google domain with a project?

Reply:
I think the answer is to create a new zone and follow: https://cloud.google.com/dns/docs/update-name-servers?